### PR TITLE
CXX-2813 KMIP delegated mode

### PR DESCRIPTION
**Pull request by adriandole**
- Sync delegated spec tests
- Bump C driver minimum version to 1.27

### CODE REVIEW

File: etc/purls.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # re-generate the SBOM JSON file!
```
Comment (kevinAlbs) [inline etc/purls.txt:6]: Regenerate the SBOM file with the new dependency.

https://github.com/mongodb/mongo-cxx-driver/blob/b78767df7defed1320e0d0c0f159871a950dc1af/etc/releasing.md#check-and-update-the-sbom-lite includes instructions.

Secrets needed to follow SBOM steps are available under `drivers/cxx-driver` in [AWS Secrets Manager](https://wiki.corp.mongodb.com/display/DRIVERS/Using+AWS+Secrets+Manager+to+Store+Testing+Secrets).

Comment (eramongodb) [reply etc/purls.txt:6]: Bump: SBOM Lite must be regenerated with new purls.

**Review by eramongodb — CHANGES_REQUESTED** [review]
Also add a "Changed" entry to CHANGELOG.md (e.g. [as was done here](https://github.com/mongodb/mongo-cxx-driver/blob/7ddaa1b1405fefa85305fef6f1fb45e35913d6d0/CHANGELOG.md?plain=1#L79)).

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Changes prior to 3.9.0 are documented as [release notes on GitHub](https://githu
 
 ## 3.11.0 [Unreleased]
 
-<!-- Will contain entries for the next minor release -->
+- Bump minimum C Driver version to [1.27.0](https://github.com/mongodb/mongo-c-driver/releases/tag/1.27.0).
```
Comment (eramongodb) [inline CHANGELOG.md:12]:
```suggestion
### Changed

- Bump minimum C Driver version to [1.27.0](https://github.com/mongodb/mongo-c-driver/releases/tag/1.27.0).
```

Missing "Changed" section. See [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) for context.

**Review by eramongodb — APPROVED** [review]
Missing "Changed" section in changelog; otherwise, LGTM.

**Pull request closed without merging**